lib/parser/livesplit_core_parser.rb

```diff
@@ -62,95 +75,99 @@ def self.parse(run, fast: false)
-            gametime: history_element_time.game_time.try(:total_seconds) || 0,
-            realtime: history_element_time.real_time.try(:total_seconds) || 0
-          }
-
-          split.realtime_history << history_element_time.real_time.try(:total_seconds) || 0
-        end
-        history_iterator.dispose
-      end
+      split[:gametime_end_ms] = (segment.personal_best_split_time.game_time.try(:total_seconds) || 0) * 1000
+      split[:gametime_duration_ms] = [0, split[:gametime_end_ms] - run_object[:gametime_duration_ms]].max
+      split[:gametime_start_ms] = split[:gametime_end_ms] - split[:gametime_duration_ms]
 
-      split.gametime_end = segment.personal_best_split_time.game_time.try(:total_seconds) || 0
-      split.gametime_duration = [0, split.gametime_end - run_object[:gametime_time]].max
-      split.gametime_start = split.gametime_end - split.gametime_duration
+      split[:gametime_best_ms] = (segment.best_segment_time.game_time.try(:total_seconds) || 0) * 1000
+      split[:gametime_skipped] = split[:gametime_duration_ms].zero?
 
-      split.gametime_best = segment.best_segment_time.game_time.try(:total_seconds) || 0
-      split.gametime_skipped = split.gametime_duration.zero?
+      if split[:gametime_duration_ms].positive?
+        split[:gametime_gold] = true if split[:gametime_duration_ms].round(5) <= split[:gametime_best_ms].try(:round, 5)
+      end
 
-      if split.gametime_duration.positive? && split.gametime_duration.round(5) <= split.gametime_best.try(:round, 5)
-        split.gametime_gold = true
-      else
-        split.gametime_gold = false
+      history_iterator = segment.segment_history.iter
+      until (history_element = history_iterator.next).nil?
+        history_element_time = history_element.time
+        split[:history] << {
+          attempt_number: history_element.index,
+          gametime_duration_ms: (history_element_time.game_time.try(:total_seconds) || 0) * 1000,
+          realtime_duration_ms: (history_element_time.real_time.try(:total_seconds) || 0) * 1000
+        }
       end
+      history_iterator.dispose
 
-      run_object[:realtime_time] += split.realtime_duration if split.realtime_duration.present?
-      run_object[:gametime_time] += split.gametime_duration if split.gametime_duration.present?
+      run_object[:realtime_duration_ms] += split[:realtime_duration_ms] if split[:realtime_duration_ms].present?
+      run_object[:gametime_duration_ms] += split[:gametime_duration_ms] if split[:gametime_duration_ms].present?
       run_object[:splits] << split
     end
 
     run.dispose
+    add_default_proc!(run_object)
     run_object
   end
+
+  def self.add_default_proc!(obj)
+    traverse_enumerable(obj)
+  end
+
+  def self.traverse_enumerable(obj)
+    obj.default_proc = ->(_hash, key) { raise KeyError, "#{key} not found" } if obj.respond_to?(:default_proc=)
+    return unless obj.respond_to?(:each)
+
+    obj.each do |value|
+      traverse_enumerable(value)
```

Comment:
Yeah, it does assume we aren't doing anything crazy with the keys, but I don't think it should be a problem since part of this is to simplify the output.